app/src/Containers/SearchContainer.tsx

```diff
@@ -294,7 +297,33 @@ const SearchContainer = () => {
     })
   }
 
-  const searchByArray = (
+  const exactSearchByArray = (
+    searchQueries: string[],
+    searchField: 'client' | 'description' | 'name' | 'role' | 'skills' | 'sector',
+  ): Projects => {
+    let results = [] as Projects
+
+    if (projects) {
+      results = projects.filter(
+        project => {
+          const anySearchQueriesMatching = searchQueries.some(searchQuery => {
+            if (typeof project[searchField] === "string") { // most fields are strings, but some are an array of strings (i.e. skills)
+              const stringSearchField = project[searchField] as string
+              return stringSearchField.toLowerCase().includes(searchQuery.toLowerCase())
+            }
+            else if (Array.isArray(project[searchField])) { // assume it's an array, ie skills
+              const arrayOfStrings = project[searchField] as string[]
+              return arrayOfStrings.some(item => item.toLowerCase().includes(searchQuery.toLowerCase()))
```

Comment:
If you use `.includes` on this line, you'll get results for JavaScript when you search for Java, which probably isn't what you want. Since this method is meant to be checking for an exact match, then we could just compare for string equality instead of includes.

You might want to keep the `.toLowerCase` conversions on both sides of the equality, in order to catch e.g. JavaScript vs Javascript - depends on how much consistency we can expect from the source data on that front :-) (Might want a .trim() too if you think there might be trailing spaces?)

Reply:
Very true Rosy, good point. At the moment though the skills are not actually a dropdown in which matches as exact as that can be found. The skills are hidden in a paragraph. From conversations with Joanna it appears this will likely stay like that. This means a search for an exact comparison match will not return enough of the wanted results. This way the search will find the "exact" search word if it is in the paragraph but you are right, it does mean a search for example for Java will also return JavaScript and React will also return React Native, but considering the goal of the quick searches we felt this was a good solution. Happy to discuss tomorrow if you want and open to other solutions of course.

Reply:
Thanks, this makes sense after the discussion this morning - think I'd been taking the 'exact search' description a bit too literally :-) Not sure what a better name would be: includesSearch, maybe?

Reply:
I changed the name of the 'exactSearchByArray' to 'searchByArray' and added a comment to the .includes line to clarify.